whitelist-helper-scripts/procLastAuthenticated.ts

```diff
@@ -0,0 +1,26 @@
+import { ethers } from "ethers";
+import { range, chunk } from "lodash";
+export default async (
+    _contract: ethers.Contract,
+    _whitelisted: object,
+    _whitelistedWLastAuthenticated: object,
+    step = 100,
+    parallel = 10
+  ) => {
+    const addresses = Object.keys(_whitelisted);
+    const latestItem = addresses.length;
+    const items = range(0, latestItem, step);
+    
+    for (let itemsChunk of chunk(items, parallel)) {
+      const ps = itemsChunk.map(async item => {
+          const psInt = addresses.slice(item,Math.min(item+step, latestItem)).map(async address => {
+            let lastAuthenticated = 
+                (await _contract.callStatic.lastAuthenticated(address));
+            _whitelistedWLastAuthenticated[address] = lastAuthenticated.toNumber();
+          });
+          console.log('Progress ... '+item+' from '+latestItem);
+          await Promise.all(psInt);
+      });
+      await Promise.all(ps);
+    }
```

Comment:
here there's no need for both step + parallel.
step was for how many events to query in each call and parallel is how many queries to create.
here you always query just 1 address every time. so you only need the parallel parameter. so you just chunk items according to parallel no need for the `range`